app/uk/gov/hmrc/incometaxproperty/repositories/ExpireAtCalculator.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.incometaxproperty.repositories
+
+import java.time.temporal.ChronoUnit
+import java.time.{Instant, Month, ZoneOffset}
+
+object ExpireAtCalculator {
```

Comment:
I think we can set TTL indexes during the creation of the collections, any specific reason we need to set this programatically? Thanks

Reply:
Each record will have a separate expiration date, its not fixed for all records